tasks/process_flakes.py

```diff
@@ -23,76 +28,120 @@ class ProcessFlakesTask(BaseCodecovTask, name=process_flakes_task_name):
 
     def run_impl(
         self,
-        _db_session,
+        _db_session: Any,
         *,
-        repo_id,
-        commit_id_list,
-        branch,
-        **kwargs,
+        repo_id: int,
+        commit_id: str,
+        **kwargs: Any,
     ):
-        repo_id = int(repo_id)
+        """
+        this task wants to iterate through uploads for a given commit that have yet to be
+        "flake processed"
+
+        for each of those uploads it wants to iterate through its test instances and
+        update existing flakes' count, recent_passes_count, fail_count, and end_date fields
+        depending on whether the test instance passed or failed
+
+        for each upload it wants to keep track of newly created flakes and keep those in a separate
+        collection than the existing flakes, so at the end it can bulk create the new flakes and
+        bulk update the existing flakes
+
+        it also wants to increment the flaky_fail_count of the relevant DailyTestRollup when it creates
+        a new flake so it keeps track of those changes and bulk updates those as well
+
+        when it's done with an upload it merges the new flakes dictionary into the existing flakes dictionary
+        and then clears the new flakes dictionary so the following upload considers the flakes created during the previous
+        iteration as existing
+        """
         log.info(
             "Received process flakes task",
-            extra=dict(repoid=repo_id, commit=commit_id_list),
+            extra=dict(repoid=repo_id, commit=commit_id),
         )
 
-        with metrics.timer("process_flakes"):
-            flake_dict = generate_flake_dict(repo_id)
+        uploads = ReportSession.objects.filter(
+            report__report_type=CommitReport.ReportType.TEST_RESULTS.value,
+            report__commit__commitid=commit_id,
+            state="processed",
+        ).all()
+
+        curr_flakes = fetch_curr_flakes(repo_id)
+        new_flakes: dict[str, Flake] = dict()
+
+        rollups_to_update: list[DailyTestRollup] = []
+
+        flaky_tests = list(curr_flakes.keys())
+
+        for upload in uploads:
+            test_instances = get_test_instances(upload, flaky_tests)
+            for test_instance in test_instances:
+                if test_instance.outcome == TestInstance.Outcome.PASS.value:
+                    flake = {**new_flakes, **curr_flakes}.get(test_instance.test_id)
```

Comment:
you are merging these two dicts into an intermediate dict you throw away immediately. in a loop for every test instance, which is a bad idea for performance.